mmf/datasets/databases/readers/feature_readers.py

```diff
@@ -79,28 +80,79 @@ def read(self, image_feat_path):
         return self.feat_reader.read(image_feat_path)
 
 
-class FasterRCNNFeatureReader:
+class MMFFeatureReader:
+    def read(self, feat_path, *args, **kwargs):
+        raise NotImplementedError(
+            "This feature reader doesn't implement a 'read' method"
+        )
+
+    @classmethod
+    def create(cls, name: str, *args, **kwargs):
+        if name == "default":
+            # Ensure backwards compatibility
+            return FeatureReader(*args, **kwargs)
+        else:
+            reader_cls = registry.get_feature_reader_class(name)
+            return reader_cls(*args, **kwargs)
+
+    @classmethod
+    def from_config_and_path(
+        cls, config: mmf_typings.DictConfig, path: str, *args, **kwargs
+    ):
+        params = {
+            "base_path": get_absolute_path(path),
+            "depth_first": config.get("depth_first", False),
+            "max_features": config.get("max_features", 100),
+        }
+        feature_reader_config = config.get("feature_reader", {})
+
+        if isinstance(feature_reader_config, str):
+            feature_reader = MMFFeatureReader.create(feature_reader_config, **params)
+        else:
+            feature_reader_name = feature_reader_config.get("type", "default")
+            feature_reader_params = feature_reader_config.get("params", {})
+            params.update(feature_reader_params)
+            feature_reader = cls.create(feature_reader_name, **params)
+
+        return feature_reader
+
+
+class FasterRCNNFeatureReader(MMFFeatureReader):
```

Comment:
Can we add all these classes to registry so we can use them by specifying in configs going forward?